xbmc/android/activity/XBMCApp.cpp

```diff
@@ -734,8 +746,24 @@ void CXBMCApp::onReceive(CJNIIntent intent)
   if (action == "android.intent.action.BATTERY_CHANGED")
     m_batteryLevel = intent.getIntExtra("level",-1);
   else if (action == "android.intent.action.DREAMING_STOPPED" || action == "android.intent.action.SCREEN_ON")
+  {
     if (HasFocus())
       g_application.WakeUpScreenSaverAndDPMS();
+  }
+  else if (action == "android.intent.action.HEADSET_PLUG" || action == "android.bluetooth.a2dp.profile.action.CONNECTION_STATE_CHANGED")
+  {
+    bool newstate;
+    if (action == "android.intent.action.HEADSET_PLUG")
+      newstate = (intent.getIntExtra("state", 0) != 0);
+    else
+      newstate = (intent.getIntExtra("android.bluetooth.profile.extra.STATE", 0) == 2);
```

Comment:
> Could we turn the 2 here into a constant with a readable name so that it's more obvious?

Not really, as the values are completely dependent on this specific intent. I can definitely add a comment, though ;)